app/assets/javascripts/discourse/lib/desktop-notifications.js.es6

```diff
@@ -0,0 +1,245 @@
+
+let primaryTab;
+let liveEnabled;
+let notificationTagName;
+let mbClientId;
+let lastAction;
+
+const focusTrackerKey = "focus-tracker";
+const seenDataKey = "seen-notifications";
+const recentUpdateThreshold = 1000 * 60 * 2; // 2 minutes
+const idleThresholdTime = 1000 * 10; // 10 seconds
+const INVITED_TYPE = 8;
+
+function init(container) {
+  liveEnabled = false;
+  requestPermission().then(function() {
+    try {
+      localStorage.getItem(focusTrackerKey);
+    } catch (e) {
+      Em.Logger.info('Discourse desktop notifications are disabled - localStorage denied.');
+      return false;
+    }
+    liveEnabled = true;
+    Em.Logger.info('Discourse desktop notifications are enabled.');
+    return true;
+  }).then(function(c) {
```

Comment:
Why do you have two `then` methods attached here? It looks like it could be together in one?